homeassistant/bootstrap.py

```diff
@@ -745,113 +744,74 @@ async def _async_resolve_domains_to_setup(
     #
     # Thankfully we are migrating away from the platform pattern
     # so this will be less of a problem in the future.
-    domains_to_setup.update(platform_integrations)
+    domains.update(platform_integrations)
 
-    # Load manifests for base platforms and platform based integrations
-    # that are defined under base platforms right away since we do not require
+    # Load manifests for base platforms right away since we do not require
     # the manifest to list them as dependencies and we want to avoid the lock
-    # contention when multiple integrations try to load them at once
-    additional_manifests_to_load = {
+    # contention when multiple integrations try to load them at once.
+    # Also load manifests for platform integrations that are defined under base platforms.
+    additional_domains = {
         *BASE_PLATFORMS,
         *chain.from_iterable(platform_integrations.values()),
     }
 
-    translations_to_load = additional_manifests_to_load.copy()
-
     # Resolve all dependencies so we know all integrations
     # that will have to be loaded and start right-away
-    integration_cache: dict[str, loader.Integration] = {}
-    to_resolve: set[str] = domains_to_setup
-    while to_resolve or additional_manifests_to_load:
-        old_to_resolve: set[str] = to_resolve
-        to_resolve = set()
-
-        if additional_manifests_to_load:
-            to_get = {*old_to_resolve, *additional_manifests_to_load}
-            additional_manifests_to_load.clear()
-        else:
-            to_get = old_to_resolve
-
-        manifest_deps: set[str] = set()
-        resolve_dependencies_tasks: list[asyncio.Task[bool]] = []
-        integrations_to_process: list[loader.Integration] = []
-
-        for domain, itg in (await loader.async_get_integrations(hass, to_get)).items():
-            if not isinstance(itg, loader.Integration):
-                continue
-            integration_cache[domain] = itg
-            needed_requirements.update(itg.requirements)
-
-            # Make sure manifests for dependencies are loaded in the next
-            # loop to try to group as many as manifest loads in a single
-            # call to avoid the creating one-off executor jobs later in
-            # the setup process
-            additional_manifests_to_load.update(
-                dep
-                for dep in chain(itg.dependencies, itg.after_dependencies)
-                if dep not in integration_cache
-            )
+    integrations_to_try_process = await loader.async_get_integrations(
+        hass, {*domains, *additional_domains}
+    )
+    # Eliminate those without valid manifest
+    integrations_to_process = {
+        domain: itg
+        for domain, itg in integrations_to_try_process.items()
+        if isinstance(itg, Integration)
+    }
+    integrations_dependencies = await Integration.resolve_multiple_dependencies(
+        hass, integrations_to_process.values()
+    )
```

Comment:
How come `Integration.resolve_multiple_dependencies` is a `staticmethod` of `Integration`, but other functionality, e.g. `loader.async_get_integrations` is not?
It's also confusing, because `Integration` has a method `resolve_dependencies`.

Maybe make this a function `loader.resolve_integration_dependencies` for symmetry? Same comment for `Integration.resolve_multiple_after_dependencies`